gcc-4.8/post-link.sh

```diff
@@ -0,0 +1,170 @@
+
+        c_runtime_obj_files_found=0
+
+        # Try locating crtXXX.o in default library search paths
+        for library_path in $(ld --verbose | grep SEARCH_DIR | sed -r 's/SEARCH_DIR\("=?([^"]*)"\);/ \1/g'); do
+            for obj_file in $C_RUNTIME_OBJ_FILES; do
+                obj_file_full_path="$library_path/$obj_file"
+                if [[ -e "$obj_file_full_path" ]]; then
+                    ln -s "$obj_file_full_path" "${PREFIX}/lib/gcc/"*/*/
+                    c_runtime_obj_files_found=1
+                fi
+            done
+            if [ $c_runtime_obj_files_found -eq 1 ]; then
+                break
+            fi
+        done
+
+        # Fallback to locating crtXXX.o with system gcc we if couldn't find it in usual places
+        if [ $c_runtime_obj_files_found -ne 1 ]; then
+            echo "Couldn't locate crtXXX.o in default library search paths. You may not have it " \
+                 "at all. It is usually packaged in libc6-dev/glibc-devel packages. We will try " \
+                 "to locate crtXXX.o with system installed gcc..."
+
+            SYSTEM_GCC=/usr/bin/gcc
+
+            if [ -e "${SYSTEM_GCC}" ]; then
+                for obj_file in $C_RUNTIME_OBJ_FILES; do
+                    obj_file_full_path=$($SYSTEM_GCC -print-file-name="$obj_file")
+                    if [[ "$obj_file_full_path" != "$obj_file" ]]; then
+                        ln -s "$obj_file_full_path" "${PREFIX}/lib/gcc/"*/*/
+                        c_runtime_obj_files_found=1
+                    fi
+                done
+            else
+                echo "There is no $SYSTEM_GCC"
+            fi
+        fi
+
+        if [ $c_runtime_obj_files_found -ne 1 ]; then
+            >&2 echo "*** Can't install the gcc package unless your system has crtXXX.o. ***"
+            exit 1
+        fi
+
+        #
+        # Linux Portability Issue #3: Compiler needs to locate system headers
+        #
+    
+        # Some distros use different system include paths than the ones this gcc binary was built for.
+        # We'll add these to the standard include path by providing a custom "specs file"
+    
+        # First create specs file from existing defaults
+        SPECS_DIR=$(echo "${PREFIX}"/lib/gcc/*/*)
+        SPECS_FILE="${SPECS_DIR}/specs"
+        gcc -dumpspecs > "${SPECS_FILE}"
+        
+        # Now add extra include paths to the specs file, one at a time.
+        # (So far we only know of one: from Ubuntu.)
+        EXTRA_SYSTEM_INCLUDE_DIRS="/usr/include/x86_64-linux-gnu"
+        
+        for INCDIR in ${EXTRA_SYSTEM_INCLUDE_DIRS}; do
+            # The following sed command will replace these two lines:
+            # *cpp:
+            # ... yada yada ...
+            #
+            # With these two lines:
+            # *cpp:
+            # ... yada yada ... -I${INCDIR}
+            sed -i ':a;N;$!ba;s|\(*cpp:\n[^\n]*\)|\1 -I'${INCDIR}'|g' "${SPECS_FILE}"
+        done
+    fi
+fi
```

Comment:
>Let's put the whole file in this block.

OK, on Darwin, the post-link script now exits immediately. (@asmeurer, Sorry for the delay; I was on vacation.)